foursight_core/react/api/auth.py

```diff
@@ -0,0 +1,204 @@
+import boto3
+import time
+from .cookie_utils import read_cookie
+from .envs import Envs
+from .jwt_utils import jwt_decode, jwt_encode
+
+
+class Auth:
+
+    def __init__(self, auth0_client_id: str, auth0_secret: str, envs: Envs):
+        self.auth0_client_id = auth0_client_id
+        self.auth0_secret = auth0_secret
+        self.envs = envs
+
+    class Cache:
+        aws_credentials = {}
+
+    def authorize(self, request: dict, env: str) -> dict:
+        """
+        Verifies that the given request is authenticated AND authorized, based on the authtoken
+        cookie (a JWT-signed-encoded value) in the given request. If so, returns the verified and
+        decoded authtoken as a dictionary. If not, returns a dictionary indicating not authorized
+        and/or not authenticated, and the basic info contained in the authtoken.
+        """
+        try:
+
+            # Read the authtoken cookie.
+
+            authtoken = read_cookie(request, "authtoken")
```

Comment:
Yeah I tend to do this, I don't need to; just easier on my eyes.

Reply:
I'm not sure there's official guidance here, but I agree with David, though maybe not for the reason he articulates (easy on the eyes). I save my reasons for actual semantic or pragmatic subtlety. Otherwise things get too visually stretched out (WIl's fear). But here there is a reason. I put the blank line for cases where you're talking about a bunch of code to follow that is a bigger block of code. It's perhaps more clear if we change it this way below. (And note that this separation you're using here, David, is replicated in this way in your code separating code blocks for easy chunking.)
```
# Swap two variables...

# Save one value
temp = foo
# Do first half of swap
foo = bar
# Do the second half
bar = foo
```
or as came up on one of Will's PRs today:
```
# I commented this out because of ...blah, blah ...

# x = 3
# y = 4
```

because if you uncomment those, you don't want to uncomment the first line, so you need separation.

Reply:
Yeah, this kind of thing is certainly idiosyncratic to the way we individually parse code with our eyes. I guess I tend to do this especially when the comments can be read to follow the algorithm, as an (initial) alternative to actually taking in the details of the code (like headings in an outline). Though happy to comply with whatever code/formatting standard we'd like.

Reply:
I think we're good here. Too much other stuff to do. Resolve this and move on.